plugins/deployment-metadata-backend/src/plugin.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+import { deploymentMetadataServiceRef } from './service';
+import { createRouter } from './service/router';
+
+/**
+ * @public
+ */
+export const deploymentMetadataPlugin = createBackendPlugin({
+  pluginId: 'deployment-metadata',
+
+  register(env) {
+    env.registerInit({
+      deps: {
+        logger: coreServices.logger,
+        rootHttpRouter: coreServices.rootHttpRouter,
+        deploymentMetadata: deploymentMetadataServiceRef,
+      },
+      async init({ logger, rootHttpRouter, deploymentMetadata }) {
+        logger.info('Creating deployment metadata service');
+        rootHttpRouter.use(
+          '/.backstage/metadata/deployment',
```

Comment:
Goal is to expose this for the frontend/other integration with a simple HTTP router. I can adjust if we want to use the `/api/{pluginId}` format, but this felt like a meta plugin.